### PR TITLE
Add check if the parent element has a scrollbar to determine scroll parent

### PR DESCRIPTION
**Pull request by donnysim**
Pretty sure `getScrollParent` tests should not be in `core` but couldn't get them to work in unit tests.

Fixes https://github.com/FezVrasta/popper.js/issues/526

### CODE REVIEW

File: packages/popper/src/utils/getScrollParent.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ export default function getScrollParent(element) {
 
   // Firefox want us to check `-x` and `-y` variations as well
   const { overflow, overflowX, overflowY } = getStyleComputedProperty(element);
-  if (/(auto|scroll)/.test(overflow + overflowY + overflowX)) {
+  if (/(auto|scroll)/.test(overflow + overflowY + overflowX) &&
+    (element.scrollWidth > element.clientWidth || element.scrollHeight > element.clientHeight)) {
```
Comment (FezVrasta) [inline packages/popper/src/utils/getScrollParent.js]: I'm a bit rusty, do `clientWidth` and `clientHeight` take in account possible paddings?

Comment (donnysim) [reply packages/popper/src/utils/getScrollParent.js]: Yes, it does.

Comment (donnysim) [reply packages/popper/src/utils/getScrollParent.js]: There's no need for a rush on this or if has big impact on positioning might as well close this, can get by by turning off overflow, but appreciate the help.

**Comment by FezVrasta** [comment]
Unfortunately this PR breaks some tests, do you have time to look at them?

**Pull request closed without merging**